homeassistant/components/elkm1/alarm_control_panel.py

```diff
@@ -149,30 +157,29 @@ def _watch_keypad(self, keypad, changeset):
             self._changed_by = self._elk.users.username(keypad.last_user)
             self.async_write_ha_state()
 
-    def _watch_area(self, area, changeset):
+    def _watch_area(self, area: Element, changeset: dict[str, Any]) -> None:
         if not (last_log := changeset.get("last_log")):
             return
         # user_number only set for arm/disarm logs
-        if not last_log.get("user_number"):
-            return
-        self._changed_by_keypad = None
-        self._changed_by_id = last_log["user_number"]
-        self._changed_by = self._elk.users.username(self._changed_by_id - 1)
-        self._changed_by_time = last_log["timestamp"]
-        self.async_write_ha_state()
+        if (user_number := last_log.get("user_number")) is not None:
```

Comment:
I think the `not last_log.get("user_number")` might have been a bug as `user_number` can be 0. Either way, using `is not None` is safer, IMO.

Reply:
Looks like a bug. Typing tends to catch these type of things.

We should reverse the condition and return early here to keep the indent level the same
```suggestion
        if (user_number := last_log.get("user_number")) is None:
            return
```